fontforge/splinefit.c

```diff
@@ -733,7 +805,223 @@ return( SplineMake3(from,to));
+	int numberOfSolutions;
+    SplinePoint *frompoint,*topoint;
+    f = 0; /* area */
+    m = 0; /* first area moment about y (along x) */
+    frompoint = from;
+    if ( from->next==NULL )
+		topoint=to;
+	else
+		topoint=from->next->to;
+	for ( ; ; frompoint = topoint->next->from, topoint = topoint->next->to ) {
+		/* normalizing transformation (chord to x axis and length 1) */
+		xa = ((frompoint->me.x-from->me.x)*ftunit.x+(frompoint->me.y-from->me.y)*ftunit.y)/ftlen;
+		ya = (-(frompoint->me.x-from->me.x)*ftunit.y+(frompoint->me.y-from->me.y)*ftunit.x)/ftlen;
+		xb = ((frompoint->nextcp.x-from->me.x)*ftunit.x+(frompoint->nextcp.y-from->me.y)*ftunit.y)/ftlen;
+		yb = (-(frompoint->nextcp.x-from->me.x)*ftunit.y+(frompoint->nextcp.y-from->me.y)*ftunit.x)/ftlen;
+		xc = ((topoint->prevcp.x-from->me.x)*ftunit.x+(topoint->prevcp.y-from->me.y)*ftunit.y)/ftlen;
+		yc = (-(topoint->prevcp.x-from->me.x)*ftunit.y+(topoint->prevcp.y-from->me.y)*ftunit.x)/ftlen;
+		xd = ((topoint->me.x-from->me.x)*ftunit.x+(topoint->me.y-from->me.y)*ftunit.y)/ftlen;
+		yd = (-(topoint->me.x-from->me.x)*ftunit.y+(topoint->me.y-from->me.y)*ftunit.x)/ftlen;
+		f += ((xb-xa)*(10*ya+6*yb+3*yc+yd)+(xc-xb)*(4*ya+6*yb+6*yc+4*yd)+(xd-xc)*(ya+3*yb+6*yc+10*yd))/20;
+		m += (280*xd*xd*yd-105*xc*xd*yd-30*xb*xd*yd-5*xa*xd*yd-45*xc*xc*yd-45*xb*xc*yd-12*xa*xc*yd-18*xb*xb*yd
+		-15*xa*xb*yd-5*xa*xa*yd+105*xd*xd*yc+45*xc*xd*yc-3*xa*xd*yc-27*xb*xc*yc-18*xa*xc*yc-27*xb*xb*yc
+		-45*xa*xb*yc-30*xa*xa*yc+30*xd*xd*yb+45*xc*xd*yb+18*xb*xd*yb+3*xa*xd*yb+27*xc*xc*yb+27*xb*xc*yb
+		-45*xa*xb*yb-105*xa*xa*yb+5*xd*xd*ya+15*xc*xd*ya+12*xb*xd*ya+5*xa*xd*ya+18*xc*xc*ya+45*xb*xc*ya
+		+30*xa*xc*ya+45*xb*xb*ya+105*xa*xb*ya-280*xa*xa*ya)/840;
+		if ( topoint==to )
+			break;
+    }
+    BasePoint aunit = (BasePoint) { BPDot(ftunit, fromunit), BPCross(ftunit, fromunit) }; /* normed direction at "from" */
+    BasePoint bunit = (BasePoint) { BPDot(BPRev(ftunit), tounit),BPCross(ftunit, tounit) }; /* normed direction at "to" */
+    if ( aunit.y < 0 ) { /* normalize aunit.y to >= 0: */
+		aunit.y = -aunit.y;
+		bunit.y = -bunit.y;
+		m = -m;
+		f = -f;
+	}
+	/* calculate the Tunni point (where the tangents at "from" and "to" interesect) */
+	bigreal aMax = 100; /* maximal value that the handle a can reach up to the Tunni point, 100 is really long */
+	bigreal bMax = 100; /* maximal value that the handle b can reach up to the Tunni point, 100 is really long */
+	sab = aunit.y*bunit.x+aunit.x*bunit.y; 
+	if (sab != 0) { /* if handles not parallel */
+		aMax = bunit.y/sab;
+		bMax = aunit.y/sab;
+		if ( aMax < 0 ) {
+				aMax = 100;
+		}
+		if ( bMax < 0 ) {
+				bMax = 100;
+		}
+	}
+	/* start approximation by solving the quartic equation */
+	sasa = aunit.y*aunit.y; /* reducing the multiplications */
+	Quartic quad;
+	if ( (aunit.x == -bunit.x && aunit.y == bunit.y) || (aunit.x == bunit.x && aunit.y == -bunit.y) ) { /* handles are parallel */
+		quad.a = 0;
+		quad.b = 0;
+		quad.c = -9*aunit.x*sasa;
```

Comment:
I can't seem to match this to the comment:

```c
/* 0 = -9*ca*sa^3 * a^3  */
/* + 6*sa^2*(4*sa+5*ca*f) * a^2 */
/* + 10*sa*((42*m-25*f)*sa-25*ca*f^2). */
```

If this were matching, wouldn't it be

```c
		quad.c = -9*aunit.x*sasa*aunit.y;
		quad.d = 6*sasa*(4*aunit.y+5*aunit.x*f);	
		quad.e = 10*((42*m-25*f)*aunit.y-25*aunit.x*f*f);
```

?

Reply:
The equation 
```
/* 0 = -9*ca*sa^3 * a^3  */
/* + 6*sa^2*(4*sa+5*ca*f) * a^2 */
/* + 10*sa*((42*m-25*f)*sa-25*ca*f^2). */
```
actually reduces to 
```
/* 0 = -9*ca*sa^2 * a^3  */
/* + 6*sa*(4*sa+5*ca*f) * a^2 */
/* + 10*((42*m-25*f)*sa-25*ca*f^2). */
```
since `sa` is a common factor.

Reply:
Ah, yep that makes sense, thanks.